DataStructures/Source/LinkedList/LinkedList.cs

```diff
@@ -47,41 +47,42 @@ public void AddAt(int index, T item)
                 return;
             }
 
-            var previous = ElementAt(index - 1, _head);
+            var previous = FindNodeAt(index - 1);
             previous.Next = new Node<T>(item, previous.Next);
             Length++;
         }
 
         public void Remove(T item)
         {
-            var previous = _head;
+            Node<T> previous = null;
+            var node = _head;
             for (int i = 0; i < Length; i++)
             {
-                var node = ElementAt(i, _head);
                 if (node.Item.Equals(item))
                 {
-                    RemoveNode(i, previous);
+                    RemoveNode(previous);
                     return;
                 }
 
                 previous = node;
+                node = node.Next;
```

Comment:
Something like below would be better? To me, it seems overcomplicated.
![image](https://cloud.githubusercontent.com/assets/9809523/26446979/22f2a562-414f-11e7-8b47-bdd85671b1ed.png)